src/model/core/fuel.jl

```diff
@@ -0,0 +1,117 @@
+
+@doc raw"""
+fuel!(EP::Model, inputs::Dict, setup::Dict)
+
+This function creates expression to account for total fuel consumption (e.g., coal, natural gas, hydrogen, etc). It also has the capability to model the piece-wise fuel consumption in part load (if data is available)
+
+***** Expressions ******
+
+The fuel consumption for power generation $vFuel_{y,t}$ is determined by power generation ($vP_{y,t}$) mutiplied by the corresponding heat rate ($Hear\_Rate_y$). 
+
+The fuel costs for power generation and start fuel for a plant $y$ at time $t$, denoted by $eCFuelOut_{y,t}$ and $eFuelStart$, is determined by fuel consumption ($vFuel_{y,t}$ and $eStartFuel$) multiplied by the fuel costs (\$/MMBTU)
+
+From above formulations, thermal generators are expected to have the same fuel consumption per generating 1 MWh electricity, regardless of minimum load or full load. However, thermal generators tend to have decreased efficiency when operating at part load, leading to higher fuel consumption per generating the same amount of electricity. To have more precise representation of fuel consumption at part load, the piecewise-linear fitting of heat input can be introduced. 
```

Comment:
I think this section would be much clearer if you mentioned at the top that it is possible to model both a constant heat rate and a varying heat rate, and then list the expressions for each case-- rather than explaining halfway through that the above expressions are only for the constant heat rate case.

Reply:
Good advice! I've modified the documentation.